### ISSUE TITLE
Dafny Go runtime (Binaries/DafnyRuntime.go) should be published as a Go module

### ISSUE THREAD

**Issue by robin-aws** [dafny-lang/dafny#494]

**Comment by robin-aws** [comment]
This is a more serious gap for Go in particular, because although we include the file `DafnyRuntime.go` in Dafny releases, but Go can't use this file to resolve symbols under `dafny` unless it is placed under a `dafny` folder under a directory in the `GOPATH`. 

It's also only part of the runtime since we now compile part of it from Dafny source code, so users need to include both files: https://github.com/dafny-lang/dafny/blob/master/Source/DafnyRuntime/DafnyRuntimeGo/DafnyRuntimeFromDafny.go

**Comment by atomb** [comment]
I've looked into this a little, and `go get` doesn't like the module in https://github.com/dafny-lang/dafny/blob/master/Source/DafnyRuntime/DafnyRuntimeGo-gomod because the `module` line in `go.mod` doesn't match the URL. My impression is that, in order to have a reasonable name for the Go module, we'd really need to put the runtime in a separate repository.

**Comment by atomb** [comment]
If we change the module name to `github.com/dafny-lang/dafny`, then it might work, except that `go get` might be confused by the `go.mod` files in the test suite.

**Comment by robin-aws** [comment]
Yes, my intention once we created that second copy was that as part of the Dafny release process, we'd copy the contents of it to github.com/dafny-lang/DafnyRuntimeGo.

**Comment by atomb** [comment]
Ah, good. I'll create that repository and add those files to it. For the next release, we can automate that.
